cocos/particle/animator/velocity-overtime.ts

```diff
@@ -159,9 +231,39 @@ export default class VelocityOvertimeModule extends ParticleModuleBase {
         if (this.needTransform) {
             Vec3.transformQuat(vel, vel, this.rotation);
         }
-        Vec3.add(p.animatedVelocity, p.animatedVelocity, vel);
+        // add linear velocity
+        const speed = this.speedModifier.evaluate(normalizedTime, pseudoRandom(p.randomSeed + VELOCITY_X_OVERTIME_RAND_OFFSET))!;
```

Comment:
It looks like previously the pseudo rand value was only used if `isCurveTwoValues` was true, via using `randSpeed`

Reply:
Not sure what this means. The randSpeed stuff was not actually used atm. tbh, I can't even remember why I commented it out but probably bc reasons... All of this came from the 3.7.3 branch I did a really long time ago so I'm kinda fuzzy on the details. It does appear to work though.